python/tritonserver/_api/_response.py

```diff
@@ -29,293 +29,37 @@
-
-    """Asyncio compatible response iterator
-
-    Response iterators are returned from model inference methods and
-    allow users to process inference responses in the order they were
-    received for a request.
-
-    """
-
-    def __init__(
-        self,
-        model: _model.Model,
-        request: TRITONSERVER_InferenceRequest,
-        user_queue: Optional[asyncio.Queue] = None,
-        raise_on_error: bool = False,
-        loop: Optional[asyncio.AbstractEventLoop] = None,
-    ) -> None:
-        """Initialize AsyncResponseIterator
-
-        AsyncResponseIterator objects are obtained from Model inference
-        methods and not instantiated directly. See `Model` documentation.
-
-        Parameters
-        ----------
-        model : Model
-            model associated with inference request
-        request : TRITONSERVER_InferenceRequest
-            Underlying C binding TRITONSERVER_InferenceRequest
-            object. Private.
-        user_queue : Optional[asyncio.Queue]
-            Optional user queue for responses in addition to internal
-            iterator queue.
-        raise_on_error : bool
-            if True response errors will be raised as exceptions.
-        loop : Optional[asyncio.AbstractEventLoop]
-            asyncio loop object
-
-        """
-
-        if loop is None:
-            loop = asyncio.get_running_loop()
-        self._loop = loop
-        self._queue = asyncio.Queue()
-        self._user_queue = user_queue
-        self._complete = False
-        self._request = request
-        self._model = model
-        self._raise_on_error = raise_on_error
-
-    def __aiter__(self) -> AsyncResponseIterator:
-        """Return async iterator. For use with async for loops.
-
-        Returns
-        -------
-        AsyncResponseIterator
-
-        Examples
-        --------
-
-        responses = server.model("test").async_infer(inputs={"fp16_input":numpy.array([[1]],dtype=numpy.float16)})
-        async for response in responses:
-            print(nummpy.from_dlpack(response.outputs["fp16_output"]))
-
-
-        """
-
-        return self
-
-    async def __anext__(self):
-        """Returns the next response received for a request
-
-        Returns the next response received for a request as an
-        awaitable object.
-
-        Raises
-        ------
-        response.error
-            If raise_on_error is set to True, response errors are
-            raised as exceptions
-        StopAsyncIteration
-            Indicates all responses for a request have been received.
-            Final responses may or may not include outputs and must be
-            checked.
-
-        """
-
-        if self._complete:
-            raise StopAsyncIteration
-        response = await self._queue.get()
-        self._complete = response.final
-        if response.error is not None and self._raise_on_error:
-            raise response.error
-        return response
-
-    def cancel(self) -> None:
```

Comment:
aren't we missing cancel in the new iterator?

Reply:
yes, added the ability to call cancel on the request. [Add ability to cancel a request from response iterator](https://github.com/triton-inference-server/core/pull/426/commits/dbbd68e3a10397ea0f164ac9db9255b3b4ef895b)